### PR TITLE
feat(kuma-cp): consumer policies on app's namespace

### PR DESCRIPTION
**Pull request by lobkovilya**
The PR partially implements [madr-52](https://github.com/kumahq/kuma/blob/180976f81dcb6a9b86c4c3fa534e11534060171d/docs/madr/decisions/052-policy-on-namespace.md) and allows applying policies on the custom namespace. 

Based on how policy looks Kuma CP adds a new label:
```yaml
kuma.io/policy-role: producer | consumer | workload-owner | system
```

Producer policies are not supported at this moment, because we can't reference `MeshService` in `to[].targetRef.kind`.

### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues -- Fix https://github.com/kumahq/kuma/issues/10163 
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/core/resources/model/resource.go [hunk 1]
```diff
@@ -435,6 +436,66 @@ func ResourceOrigin(rm ResourceMeta) (mesh_proto.ResourceOrigin, bool) {
...
+		hasFrom = true
+	}
+
+	if hasTo && !hasFrom {
+		// todo(lobkovilya): detect if the policy is a producer policy when they're supported
+		return mesh_proto.ConsumerPolicyRole
+	} else {
+		return mesh_proto.WorkloadOwnerPolicyRole
+	}
+}
```
Comment (Icarus9913) [inline pkg/core/resources/model/resource.go:490]: Do we need to define a `default` return? 
It seems like `if hasTo == false && hasFrom == false`, we'll return an empty string "" ?

File: pkg/plugins/policies/meshtimeout/api/v1alpha1/validator.go [hunk 2]
```diff
@@ -18,19 +20,30 @@ func (r *MeshTimeoutResource) validate() error {
...
+				common_api.MeshService,
+				common_api.MeshServiceSubset,
+				common_api.MeshHTTPRoute,
+			},
+			GatewayListenerTagsAllowed: true,
+		})
+	default:
+		return mesh.ValidateTargetRef(targetRef, &mesh.ValidateTargetRefOpts{
+			SupportedKinds: []common_api.TargetRefKind{
+				common_api.Mesh,
```
Comment (Automaat) [inline pkg/plugins/policies/meshtimeout/api/v1alpha1/validator.go:40]: shouldn't we allow targeting `MeshHttpRoute`? Or will it be changed in the future with full support for policies on namespace?

File: pkg/core/resources/model/resource.go (hunk 1 above)
Comment (Automaat) [reply pkg/core/resources/model/resource.go:490]: why? If I read it correctly, we will return either `mesh_proto.ConsumerPolicyRole` or `mesh_proto.WorkloadOwnerPolicyRole` there are no other return statements in this function

**Review by Automaat — APPROVED** [review]

File: pkg/core/resources/model/resource.go (hunk 1 above)
Comment (Icarus9913) [reply pkg/core/resources/model/resource.go:490]: Oh, you're right. 
My fault, my head was foggy.

File: pkg/plugins/policies/meshtimeout/api/v1alpha1/validator.go (hunk 2 above)
Comment (lobkovilya) [reply pkg/plugins/policies/meshtimeout/api/v1alpha1/validator.go:40]: TBH I don't want to allow MeshHTTPRoute in top-level targetRef of namespace-scoped policies to deprecated in the near future with https://github.com/kumahq/kuma/issues/10247.

**Pull request merged**